File: src/execgo_playground/schema.py
from __future__ import annotations

from pathlib import Path

from .models import (
    BenchmarkResult,
    BenchmarkRunRequest,
    ChaosProfile,
    ScenarioSpec,
    StandardPlan,
    TimelineEvent,
)
from .utils import ensure_dir, write_json


SCHEMA_MODELS = {
    "benchmark-result.schema.json": BenchmarkResult,
    "benchmark-run-request.schema.json": BenchmarkRunRequest,
    "chaos-profile.schema.json": ChaosProfile,
    "scenario.schema.json": ScenarioSpec,
    "standard-plan.schema.json": StandardPlan,
    "timeline-event.schema.json": TimelineEvent,
}


def export_schemas(output_dir: str | Path) -> list[Path]:
    out = ensure_dir(output_dir)
    written: list[Path] = []
    for filename, model in SCHEMA_MODELS.items():
        path = out / filename
        write_json(path, model.model_json_schema())
        written.append(path)
    return written

File: src/execgo_playground/schema.ts
import path from "node:path";
import { readJson, repoRoot, writeJson } from "./utils.js";

const SCHEMA_FILES = [
  "benchmark-result.schema.json",
  "benchmark-run-request.schema.json",
  "chaos-profile.schema.json",
  "scenario.schema.json",
  "standard-plan.schema.json",
  "timeline-event.schema.json",
];

export async function exportSchemas(outputDir: string): Promise<string[]> {
  const sourceDir = path.join(repoRoot(), "shared", "spec");
  const written: string[] = [];
  for (const filename of SCHEMA_FILES) {
    const target = path.join(outputDir, filename);
    const source = path.join(sourceDir, filename);
    const schema = await readJson(source);
    await writeJson(target, schema);
    written.push(target);
  }
  return written;
}
